src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -324,32 +335,45 @@ void TileEmitter::validate_arguments(const std::vector<size_t> &in,
                                      const std::vector<size_t> &out,
                                      const std::vector<size_t> &pool,
                                      const std::vector<size_t> &gpr) const {
-    if (in.size() != 2)
-        IE_THROW() << "TileEmitter got invalid number of inputs. Expected 2, got " << in.size();
-    if (!out.empty())
-        IE_THROW() << "TileEmitter got invalid number of outputs. Expected 0" << " , got " << out.size();
+    if (in.size() != 1)
+        IE_THROW() << "TileEmitter got invalid number of inputs. Expected 1, got " << in.size();
+    if (out.size() != io_dims.size())
+        IE_THROW() << "TileEmitter got invalid number of outputs. Expected " << io_dims.size() << " , got " << out.size();
+}
+
+void TileEmitter::emit_body(const std::vector<size_t>& vec_pool, const std::vector<size_t>& gpr_pool) const {
+    for (auto& code : body)
+        code.first->emit_code(code.second.first, code.second.second, vec_pool, gpr_pool);
+}
+
+void TileEmitter::emit_ptr_increments(const std::vector<Reg64>& data_ptr_regs) const {
+    for (size_t i = 0; i < num_inputs; i++) {
+        // those with dims == 1 will be broadcasted, hence don't require increment
+        if (io_dims[i] != 1)
+            h->add(data_ptr_regs[i], increment * sizeof(float));
+    }
+    for (size_t i = num_inputs; i < num_inputs + num_outputs; i++)
+        h->add(data_ptr_regs[i], increment * sizeof(float));
 }
 
 void TileEmitter::emit_impl(const std::vector<size_t>& in,
                             const std::vector<size_t>& out,
                             const std::vector<size_t>& vec_pool,
                             const std::vector<size_t>& gpr_pool,
                             const ov::intel_cpu::emitter_context *emit_context) const {
-    const size_t inc = in[0];
-    Reg64 work_amount = Reg64(static_cast<int>(in[1]));
+    Reg64 work_amount = Reg64(static_cast<int>(in[0]));
+    std::vector<Reg64> data_ptr_regs(out.size());
+    std::transform(out.begin(), out.end(), data_ptr_regs.begin(), [](size_t idx){return Reg64(static_cast<int>(idx));});
```

Comment:
Seems like we already need helper function for that)

Reply:
Seriously, it's a bit tedious to cast from size_t every time, maybe we should consider to make all emit_impl params `int` in the future